src/shared/compatibility.js

```diff
@@ -205,23 +205,53 @@ const hasDOM = typeof window === 'object' && typeof document === 'object';
                               globalScope.Promise.prototype.finally)) {
     return;
   }
-  globalScope.Promise = require('core-js/fn/promise');
+  globalScope.Promise = require('core-js/es/promise/index');
+})();
+
+// Support: IE
+(function checkURL() {
+  if (typeof PDFJSDev !== 'undefined' && PDFJSDev.test('IMAGE_DECODERS')) {
+    // The current image decoders don't use the `URL` constructor, so it
+    // doesn't need to be polyfilled for the IMAGE_DECODERS build target.
+    return;
+  }
+  if (typeof PDFJSDev !== 'undefined' && !PDFJSDev.test('GENERIC')) {
+    // The `URL` constructor is assumed to be available in the extension builds.
+    return;
+  }
+
+  let isURLSupported = false;
```

Comment:
Given that `core-js` should have its own [feature detection](https://github.com/zloirock/core-js/blob/master/packages/core-js/internals/native-url.js), could/should all of this code simply be replaced by `require('core-js/web/url');` instead?